extension/demo/src/bg/background.js

```diff
@@ -1,66 +1,85 @@
-    if (!results[STORAGE_KEY]) {
-      chrome.tabs.create({
-        url: `${API_BASE}/oauth/ext`,
-        active: true
-      })
-    }
-  })
+async function onPopupTriggerAuth () {
+  const accessToken = await storage.get(ACCESS_TOKEN)
+  if (!accessToken) {
+    chrome.tabs.create({
+      url: `${API_BASE_WEB}/oauth/ext`,
+      active: true
+    })
+  }
 }
 
 function onPopupOpenApps () {
   chrome.tabs.create({
-    url: `${API_BASE}/apps`,
+    url: `${API_BASE_WEB}/apps`,
     active: true
   })
 }
 
 async function onToken (data) {
   const { accessToken, subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY] = {
-    auth: {
-      accessToken: accessToken,
-      subscription: subscription
-    }
-  }
-  await chrome.storage.local.set(payload)
-  const isAuthenticated = !chrome.runtime.lastError
+  const isAuthenticated = await storage.set({ accessToken, subscription })
+  storage.get(ACCESS_TOKEN).then(console.log)
   sendMessage('popup-set-view', {
     isAuthenticated
   })
+  updatePayload(accessToken)
+
   return {
     isAuthenticated
   }
 }
 
+function timeToLive (expiresAt) {
+  return expiresAt - (Math.floor(Date.now() / 1000) + 3600)
+}
+
+async function updatePayload (accessToken) {
+  const payload = await storage.get(PAYLOAD)
+  if (!payload) {
+    const subscriptionStatus = await api.fetchSubscriptionStatus(accessToken)
+    console.log(subscriptionStatus)
+    if (subscriptionStatus.hasOwnProperty(PAYLOAD)) {
+      storage.set(subscriptionStatus)
+    }
+  } else {
+    const ttl = await storage.get(TTL)
+    if (ttl) {
+      let timeout = timeToLive(ttl)
+      console.log(timeout)
+      setTimeout(updatePayload.bind(null, accessToken), timeout)
+    }
+  }
+}
+
 async function onSubscription (data) {
   const { subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY]['auth']['subscription'] = subscription
-  await chrome.storage.local.set(payload)
-  chrome.storage.local.get([STORAGE_KEY], results => {
-    console.log(results)
-    return results
-  })
+  await storage.set({ subscription })
 }
 
 async function onPopupCheckAuth () {
-  await chrome.storage.local.get([STORAGE_KEY], results => {
-    sendMessage('popup-set-view', {
-      isAuthenticated: !!results[STORAGE_KEY]
-    })
+  const accessToken = await storage.get(ACCESS_TOKEN)
+  sendMessage('popup-set-view', {
+    isAuthenticated: !!accessToken
   })
 }
 
-addListener('token', onToken)
-addListener('subscription', onSubscription)
-addListener('popup-check-auth', onPopupCheckAuth)
-addListener('popup-trigger-auth', onPopupTriggerAuth)
-addListener('popup-open-apps', onPopupOpenApps)
+;(async () => {
+  addListener('token', onToken)
+  addListener('subscription', onSubscription)
+  addListener('popup-check-auth', onPopupCheckAuth)
+  addListener('popup-trigger-auth', onPopupTriggerAuth)
+  addListener('popup-open-apps', onPopupOpenApps)
+
+  const accessToken = storage.get(ACCESS_TOKEN)
```

Comment:
```suggestion
  const accessToken = await storage.get(ACCESS_TOKEN)
```

`storage.get` returns a promise, so you need to have an `await` here

Reply:
Ah, so await works then, I am confusion, the code is since changed to then() but would be cleaner like this